docs/httplug.md

```diff
@@ -0,0 +1,52 @@
+
+See the (tutorial)[tutorial.md] for a concrete example.
+
+
+## Httplug implementations
+
+Httplug implementations typically are either HTTP clients of their own, or they are adapters wrapping existing clients like Guzzle 6. In the later case, they will depend on the required client implementation, you should only need to require the adapter and not the actual client.
+
+Clients usually also provide a (message factory)[message-factory.md] to allow you to create requests. See (packagist)[https://packagist.org/providers/php-http/client-implementation] for the full list of implementations.
+
+Note: Until Httplug 1.0 becomes stable, we will focus on the Guzzle6 adapter.
+
+## Usage in a project
+
+When installing in an application or a non-reusable package, you should require a concrete client implementation. The client will in turn depend on `php-http/httplug`.
+
+A few things should be taken into consideration before choosing an adapter:
+
+- It is possible that some other dependency already has an HTTP Client requirement like Guzzle 6. It can be confusing to have more than one HTTP Client installed, so always check your other requirements and choose an adapter based on that.
+- Some clients support parallel requests, others only emulate them. If parallel requests are needed, use one which supports it.
+
+
+## Installation in a reusable package
+
+In many cases, packages are designed to be reused from the very beginning. For example, API clients are usually used in other packages/applications, not on their own.
+
+In these cases, they should **not rely on a concrete implementation** (like Guzzle 6), but only require any implementation of Httplug. Httplug uses the concept of virtual packages. Instead of depending on only the interfaces, which would be missing an implementation, or depending on one concrete implementation, you should depend on the virtual package `php-http/client-implementation`. There is no package with that name, but all clients and adapters implementing Httplug declare that they provide this virtual package.
+
+You need to edit the `composer.json` of your package to add the virtual package. For development (installing the package standalone, running tests), add a concrete implementation in the `require-dev` section to make the project installable:
+
+``` json
+...
+"require": { 
+    "php-http/client-implementation": "^1.0" 
+}, 
+"require-dev": { 
+    "php-http/guzzle6-adapter": "^1.0" 
+},
+...
+```
+
+If injecting the client instance into the code of the reusable package is not convenient, you can use the (Discovery)[discovery.md] system to free the user of your package from having to instantiate the client.
```

Comment:
I would add an extra note here: "However, this way they are losing the possibility to configure it."